internal/plugins/helm/v1/chartutil/chart.go

```diff
@@ -205,14 +189,24 @@ func fetchChart(destDir string, opts CreateOptions) (*golang.Options, *chart.Cha
 		opts.GVK.Kind = strcase.ToCamel(chartName)
 	}
 
-	r := &golang.Options{}
-	r.Namespaced = true
-	r.Domain = opts.Domain
-	r.Group = opts.GVK.Group
-	r.Version = opts.GVK.Version
-	r.Kind = opts.GVK.Kind
+	return opts.NewResource(cfg), chart, nil
+}
 
-	return r, chart, nil
+func (opts CreateOptions) NewResource(cfg config.Config) resource.Resource {
+	ro := &golang.Options{}
```

Comment:
[`golang.Options.NewResource()`](https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/options.go#L137) configures a resource more thoroughly. That code could perhaps be copied to the operator-sdk internally or made more generic upstream (this was the case previously but was refactored for compartmentalization reasons I think).

Reply:
HI @joelanford, 

Yes, it is not good and it was discussed already. We have todos in the SDK code to fix that. Changes were made in KB side and we toke to long to align it here and then, we agreed to move forward with todos and solve it in a follow ups since kb should provide the funcs to create a new resource and validate it according to its rules outside of the golang.options. 

See: https://github.com/kubernetes-sigs/kubebuilder/issues/1973 and https://github.com/kubernetes-sigs/kubebuilder/pull/1974. Also, see https://github.com/kubernetes-sigs/kubebuilder/pull/1986 and https://github.com/kubernetes-sigs/kubebuilder/pull/1986#issuecomment-775731617.

Reply:
Just as an addition to Camila and Eric's comments, this is made more generic with the plugin phase 1.5 implementation. As plugins are chainable, the resource is created by the CLI and injected into the plugin which no longer has the responsability to create the resource itself. #4581 shows how SDK would be aligned to kubebuilder after 1.5 implementation, and there we no longer have this dependency.